ui/src/app/connect/dialog/start-all-adapters/start-all-adapters-dialog.component.ts

```diff
@@ -0,0 +1,141 @@
+import {
+    AdapterDescriptionUnion,
+    AdapterService,
+} from '@streampipes/platform-services';
+
+@Component({
+    selector: 'sp-start-all-adapters-dialog',
+    templateUrl: './start-all-adapters-dialog.component.html',
+    styleUrls: ['./start-all-adapters-dialog.component.scss'],
+})
+export class StartAllAdaptersDialogComponent implements OnInit {
+    @Input()
+    adapters: AdapterDescriptionUnion[];
+
+    adaptersToModify: AdapterDescriptionUnion[];
+    installationStatus: any;
+    installationFinished: boolean;
+    page: string;
+    nextButton: string;
+    installationRunning: boolean;
+
+    @Input()
+    action: boolean;
+
+    constructor(
+        private dialogRef: DialogRef<StartAllAdaptersDialogComponent>,
+        private adapterService: AdapterService,
+    ) {
+        this.adaptersToModify = [];
+        this.installationStatus = [];
+        this.installationFinished = false;
+        this.page = 'preview';
+        this.nextButton = 'Next';
+        this.installationRunning = false;
+    }
+
+    ngOnInit() {
+        this.getAdaptersToModify();
+        if (this.adaptersToModify.length === 0) {
+            this.nextButton = 'Close';
+            this.page = 'installation';
+        }
+    }
+
+    close(refreshAdapters: boolean) {
+        this.dialogRef.close(refreshAdapters);
+    }
+
+    next() {
+        if (this.page === 'installation') {
+            this.close(true);
+        } else {
+            this.page = 'installation';
+            this.initiateInstallation(this.adaptersToModify[0], 0);
+        }
+    }
+
+    getAdaptersToModify() {
+        this.adapters.forEach(adapter => {
+            this.adaptersToModify.push(adapter);
+        });
+    }
+
+    initiateInstallation(adapter: AdapterDescriptionUnion, index) {
+        this.installationRunning = true;
+        this.installationStatus.push({
+            name: adapter.name,
+            id: index,
+            status: 'waiting',
+        });
+        if (this.action) {
+            this.startAdapter(adapter, index);
+        } else {
+            this.stopAdapter(adapter, index);
+        }
+    }
+
+    startAdapter(adapter: AdapterDescriptionUnion, index) {
+        this.adapterService
+            .startAdapter(adapter)
+            .subscribe(data => {
+                this.installationStatus[index].status = data.success
+                    ? 'success'
+                    : 'error';
+            })
+            .add(() => {
+                if (index < this.adaptersToModify.length - 1) {
+                    index++;
+                    this.initiateInstallation(
+                        this.adaptersToModify[index],
+                        index,
+                    );
+                } else {
+                    this.nextButton = 'Close';
+                    this.installationRunning = false;
+                }
+            });
+    }
+
+    stopAdapter(adapter: AdapterDescriptionUnion, index) {
```

Comment:
Can we simplify both methods as they are very similar?
E.g., can we introduce a method which takes the observable and the nextButton string as an argument and performs the operation?

Reply:
Sure, will look into it!